### DISCUSSION TITLE
エディターのメニューからパッケージ化したい

### DISCUSSION THREAD

**Discussion by botamochi0x12** [orukRed/tyranosyntax#201] [Ideas]
## 概要

VSCode上の操作でパッケージ化できるようにする。
例えば、あるノベルゲームのプロジェクトのアセットを頒布可能なファイルにコンパイル・ビルドする。

## 提案

パッケージ化について複数種類の方法で実現できる：

- `data/` ディレクトリ以下を `asar` 形式に圧縮し、内部的に Electron で `asar` を読み出す実行可能ファイルを作る [^fn:tyranostudio]
- Electron の参照するディレクトリをビルドするツールを使い、パッケージ化された実行可能ファイルを作る[^fn:skynovel][^fn:tyranoscript-packager]

[^fn:tyranostudio]: [`TyranoScript`](https://github.com/ShikemokuMK/tyranoscript.git) プロジェクトさんの `TyranoStudio` を参考にしました
[^fn:skynovel]: `famibee` さんの [`SkyNovel`](https://github.com/famibee/SKYNovel-vscode-extension?tab=readme-ov-file#%E3%82%B3%E3%83%BC%E3%83%87%E3%82%A3%E3%83%B3%E3%82%B0%E8%A3%9C%E5%8A%A9) を参考にしました。特に [SKYNovel_sample](https://github.com/famibee/SKYNovel_sample?tab=readme-ov-file#%E9%85%8D%E5%B8%83%E7%94%A8%E3%82%A2%E3%83%97%E3%83%AA%E3%83%91%E3%83%83%E3%82%B1%E3%83%BC%E3%82%B8%E4%BD%9C%E6%88%90%E3%82%A2%E3%83%97%E3%83%AA%E7%89%88) は UI も参考になります
[^fn:tyranoscript-packager]: `fuji44` さんの [`tyranoscript-packager`](https://github.com/fuji44/tyranoscript-packager) を参考にしました

**Comment by botamochi0x12** [comment]
Issue のタイトルで示しているようなことが実現できれば便利だと 私 (@botamochi0x12) は思っています
@orukRed さんはどう思われますか？

**Reply by orukRed** [reply]
@botamochi0x12 
ご提案ありがとうございます！
個人的な見解を以下に記載します。

結論から申し上げますと、実装すること自体で便利になると思います。
ただし、以下の懸念があります。 👀 
- 公式にもパッケージング機能は存在しているため、公式のものでない旨の警告を添えて置きたいと考えています。
- もし作るのであれば、公式のパッケージング機能との差別化を図りたいです。
  - 個人的には「公式のものあるしそれ使えばいいかな？」と考えていたので追加したい機能などあればそちらも共有いただけると嬉しいです 🙌 
- 本拡張機能でパッケージングした場合、ノベルゲームコレクションでの公開に問題がないか（なさそうな気はしているのですが、念のため）
- 実装することで利用する方が「どっち使えばいいのかな？」と迷ってしまわないか（これは杞憂な気もしていますが、同じく念のため）

**Reply by botamochi0x12** [reply]
ご回答いただきありがとうございます😊

共通して公式パッケージャーとの違いによる利点・欠点を明らかにしなければならないということですね
この観点は抜けていました💦

考えをまとめてから本文に追記します🖊
